x-pack/plugins/apm/public/components/fleet_integration/apm_agents/runtime_attachment/discovery_rule.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiBadge,
+  EuiPanel,
+  DraggableProvidedDragHandleProps,
+} from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { Operation } from '.';
+
+interface Props {
+  id: string;
+  order: number;
+  operation: string;
+  type: string;
+  probe: string;
+  providedDragHandleProps?: DraggableProvidedDragHandleProps;
+  onDelete: (discoveryItemId: string) => void;
+  onEdit: (discoveryItemId: string) => void;
+  operationTypes: Operation[];
+}
+
+export function DiscoveryRule({
+  id,
+  order,
+  operation,
+  type,
+  probe,
+  providedDragHandleProps,
+  onDelete,
+  onEdit,
+  operationTypes,
+}: Props) {
+  const operationTypesLabels = useMemo(() => {
+    return operationTypes.reduce((acc, current) => {
+      return {
+        ...acc,
+        [current.operation.value]: {
+          label: current.operation.label,
+          types: current.types.reduce((memo, { value, label }) => {
+            return { ...memo, [value]: label };
+          }, {}),
+        },
+      };
+    }, {} as { [operationValue: string]: { label: string; types: { [typeValue: string]: string } } });
```

Comment:
Instead of the `as` I think you can do `reduce<{ [operationValue: string]: { label: string; types: { [typeValue: string]: string } } }>`